spring-cloud-stream/src/main/java/org/springframework/cloud/stream/error/AbstractMessageChannelErrorConfigurer.java

```diff
@@ -0,0 +1,273 @@
+			BridgeHandler errorBridge = new BridgeHandler();
+			errorBridge.setOutputChannel(defaultErrorChannel);
+			errorChannel.subscribe(errorBridge);
+			String errorBridgeHandlerName = getErrorBridgeName(consumerBinding.getDestination(), group, consumerProperties);
+			beanFactory.registerSingleton(errorBridgeHandlerName, errorBridge);
+			beanFactory.initializeBean(errorBridge, errorBridgeHandlerName);
+		}
+		errorInfrastructureMap.put(consumerBinding.getDestination().getName(),new ErrorInfrastructure(errorChannel,recoverer,handler));
+	}
+
+	@Override
+	public void destroy(Binding<MessageChannel> binding, String group, C consumerProperties) {
+		try {
+			MessageProducerBinding consumerBinding = (MessageProducerBinding)binding;
+			String recoverer = getErrorRecovererName(consumerBinding.getDestination(), group, consumerProperties);
+			if (getApplicationContext().containsBean(recoverer)) {
+				((DefaultSingletonBeanRegistry) getApplicationContext().getBeanFactory()).destroySingleton(recoverer);
+			}
+			String errorChannelName = errorsBaseName(consumerBinding.getDestination(), group, consumerProperties);
+			String errorMessageHandlerName = getErrorMessageHandlerName(consumerBinding.getDestination(), group, consumerProperties);
+			String errorBridgeHandlerName = getErrorBridgeName(consumerBinding.getDestination(), group, consumerProperties);
+			MessageHandler bridgeHandler = null;
+			if (getApplicationContext().containsBean(errorBridgeHandlerName)) {
+				bridgeHandler = getApplicationContext().getBean(errorBridgeHandlerName, MessageHandler.class);
+			}
+			MessageHandler handler = null;
+			if (getApplicationContext().containsBean(errorMessageHandlerName)) {
+				handler = getApplicationContext().getBean(errorMessageHandlerName, MessageHandler.class);
+			}
+			if (getApplicationContext().containsBean(errorChannelName)) {
+				SubscribableChannel channel = getApplicationContext().getBean(errorChannelName, SubscribableChannel.class);
+				if (bridgeHandler != null) {
+					channel.unsubscribe(bridgeHandler);
+					((DefaultSingletonBeanRegistry) getApplicationContext().getBeanFactory())
+							.destroySingleton(errorBridgeHandlerName);
+				}
+				if (handler != null) {
+					channel.unsubscribe(handler);
+					((DefaultSingletonBeanRegistry) getApplicationContext().getBeanFactory())
+							.destroySingleton(errorMessageHandlerName);
+				}
+				((DefaultSingletonBeanRegistry) getApplicationContext().getBeanFactory())
+						.destroySingleton(errorChannelName);
+			}
+			errorInfrastructureMap.remove(consumerBinding.getDestination().getName());
+		}
+		catch (IllegalStateException e) {
+			// context is shutting down.
+		}
+	}
+
+
+	/**
+	 * Binders can return an {@link ErrorMessageStrategy} for building error messages; binder
+	 * implementations typically might add extra headers to the error message.
+	 * @return the implementation - may be null.
+	 */
+	protected ErrorMessageStrategy getErrorMessageStrategy() {
+		return null;
```

Comment:
Following on the previous comment, could it even be simpler if we return an instance of the ```DefaultErrorMessageStrategy``` here? Individual Binders can still override, but we can avoid all that null checking.
That said, I still believe the 'ErrorMessageSendingRecoverer' constructors need to be addressed , but that would be a separate and unrelated issue